app/controllers/CategoryGuidanceController.scala

```diff
@@ -44,40 +46,49 @@ class CategoryGuidanceController @Inject() (
   view: CategoryGuidanceView,
   auditService: AuditService,
   categorisationService: CategorisationService,
-  navigator: Navigator,
-  goodsRecordConnector: GoodsRecordConnector
+  goodsRecordsConnector: GoodsRecordConnector,
+  navigator: Navigator
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController
     with I18nSupport
     with Logging {
 
   def onPageLoad(recordId: String): Action[AnyContent] =
     (identify andThen profileAuth andThen getData andThen requireData).async { implicit request =>
-      categorisationService
-        .requireCategorisation(request, recordId)
-        .flatMap { userAnswers =>
-          val recordCategorisations = userAnswers.get(RecordCategorisationsQuery)
-          val categorisationInfo    = recordCategorisations.flatMap(_.records.get(recordId))
-          val scenario              = categorisationInfo.map(Scenario.getRedirectScenarios)
-          scenario match {
-            case Some(Category1NoExemptions | StandardNoAssessments) =>
-              CategoryRecord
-                .build(userAnswers, request.eori, recordId)
-                .map { categoryRecord =>
-                  goodsRecordConnector
-                    .updateCategoryAndComcodeForGoodsRecord(request.eori, recordId, categoryRecord)
-                    .map { _ =>
-                      Redirect(routes.CategorisationResultController.onPageLoad(recordId, scenario.get).url)
-                        .removingFromSession(dataUpdated, pageUpdated, dataRemoved)
-                    }
+      // Check if the commodity code is expired
+      goodsRecordsConnector
+        .getRecord(eori = request.eori, recordId = recordId)
+        .flatMap { record =>
+          if (isCommCodeExpired(record.comcodeEffectiveToDate)) {
```

Comment:
Just a note this will have to move when merged into categorisation
(I'm assuming you'll merge first :) )